stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java

```diff
@@ -199,8 +207,58 @@ public static HttpResource findResource(HTTPServicesRegistry servicesRegistry,
                     // application deal with the value.
                 }
             }
-            bValues[i] = new BString(argumentValue);
+            bValues[i + 2] = new BString(argumentValue);
+        }
+
+        if (signatureParams.getEntityBody() == null) {
+            return bValues;
+        }
+        try {
+            bValues[bValues.length - 1] = populateAndGetEntityBody(httpResource, inRequest, inRequestEntity,
+                    signatureParams.getEntityBody().getVarType());
+        } catch (BallerinaException ex) {
+            httpCarbonMessage.setProperty(HttpConstants.HTTP_STATUS_CODE, 400);
```

Comment:
Better to have a constant which reflects the meaning of the HTTP status code 400 here, for better readability and maintainability.

Reply:
Ack